airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -2054,6 +2054,7 @@ def create_default_paginator(
         config: Config,
         *,
         url_base: str,
+        extractor_model: Optional[Union[CustomRecordExtractorModel, DpathExtractorModel]] = None,
```

Comment:
It may seem weird that this is an `Optional` component since for non-custom component manifests `RecordSelector` and `RecordExtractor` are required components. However, a custom retriever that instantiates a `DefaultPaginator` won't be able to easily pass in the extractor model so it is a little easier for this to be optional and default to the existing behavior.

I know it sort of sucks, but wanted to call it out and if anyone has a better solution, I'm happy to defer to that